tokio-threadpool/src/lib.rs

```diff
@@ -1331,6 +1379,7 @@ impl Worker {
         let notify = Arc::new(Notifier {
             inner: Arc::downgrade(&self.inner),
         });
+        let mut sender = Sender { inner: self.inner.clone() };
```

Comment:
Just a thought: the executor is available a thread-local at this point. Maybe instead of the clone + threading the sender through, it can be pulled from the thread-local as needed?

I haven't considered the implications in depth either way.

Reply:
This only happens once per worker thread so doens't seem worth worrying over.

Reply:
That's fine